PyInstaller/cli/opts.py

```diff
@@ -0,0 +1,448 @@
+        'This option is automatically set to true on Windows if the first script is a `.pyw` file. '
+        'This option is ignored on *NIX systems.'
+    )(cmd)
+    cmd = optgroup.option(
+        '-i',
+        '--icon',
+        'icon_file',
+        metavar='<FILE.ico or FILE.exe,ID or FILE.icns or NONE>',
+        help='FILE.ico: apply the icon to the built executable if running on Windows. FILE.exe,ID: extract the '
+        'icon with the identifier ID from the specified executable and apply it to the built executable '
+        'if running on Windows. FILE.icns: apply the icon to the .app bundle on macOS. NONE: don\'t apply '
+        'any icon, thereby making the OS show a default icon. (default: apply PyInstaller\'s icon)'
+    )(cmd)
+    cmd = optgroup.option(
+        '--disable-windowed-traceback',
+        'disable_windowed_traceback',
+        default=False,
+        is_flag=True,
+        help='Disable traceback dump of unhandled exception in windowed (noconsole) mode '
+        '(Windows and macOS only), and instead display a message that this feature is disabled.'
+    )(cmd)
+    cmd = optgroup('Windows and macOS specific options')(cmd)
+
+    # Group: Where to search
+
+    cmd = optgroup.option(
+        '--additional-hooks-dir',
+        'hookspath',
+        multiple=True,
+        type=click.Path(),
+        help='Additional path to search for hooks. This option can be used multiple times.'
+    )(cmd)
+    cmd = optgroup.option(
+        '-p',
+        '--paths',
+        multiple=True,
+        type=click.Path(),
+        help='A path to search for imports (like using PYTHONPATH). '
+        f'Multiple paths are allowed, separated by ``{os.pathsep}``, or '
+        'use this option multiple times. Equivalent to '
+        'supplying the ``pathex`` argument in the spec file.'
+    )(cmd)
+    cmd = optgroup('Where to search')(cmd)
+
+    # Group: What to bundle
+
+    cmd = optgroup.option(
+        '--runtime-hook',
+        'runtime_hooks',
+        multiple=True,
+        type=click.File(),
+        help='Path to a custom runtime hook file. A runtime hook is code that '
+        'is bundled with the executable and is executed before any '
+        'other code or module to set up special features of the runtime '
+        'environment. This option can be used multiple times.'
+    )(cmd)
+    cmd = optgroup.option(
+        '--collect-metadata-recursive',
+        multiple=True,
+        metavar='PACKAGENAME',
+        help='Recursively collect all metadata from the specified package and it\'s dependencies. '
+        'This option can be use multiple times.'
+    )(cmd)
+    cmd = optgroup.option(
+        '--collect-metadata',
+        multiple=True,
+        metavar='PACKAGENAME',
+        type=str,
+        help='Collect all metadata from the specified package. '
+        'This option can be use multiple times.'
+    )(cmd)
```

Comment:
I know we inherited these but can we not munge packages with distributions. 

```suggestion
    cmd = optgroup.option(
        '--collect-metadata-recursive',
        multiple=True,
        metavar='DISTRIBUTIONNAME',
        help='Recursively collect all metadata from the specified distribution and it\'s dependencies. '
        'This option can be use multiple times.'
    )(cmd)
    cmd = optgroup.option(
        '--collect-metadata',
        multiple=True,
        metavar='DISTRIBUTIONNAME',
        type=str,
        help='Collect all metadata from the specified distribution. '
        'This option can be use multiple times.'
    )(cmd)
```